packages/contracts/contracts/LenderCommitmentForwarder/extensions/CommitmentRolloverLoan.sol

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+// Contracts
+import "@openzeppelin/contracts-upgradeable/utils/AddressUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/IERC20Upgradeable.sol";
+
+// Interfaces
+import "../../interfaces/ITellerV2.sol";
+
+import "../../interfaces/ILenderCommitmentForwarder.sol";
+import "../../interfaces/ICommitmentRolloverLoan.sol";
+contract CommitmentRolloverLoan is ICommitmentRolloverLoan{
+    using AddressUpgradeable for address;
+
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    ITellerV2 public immutable TELLER_V2;
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    ILenderCommitmentForwarder public immutable LENDER_COMMITMENT_FORWARDER;
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor(address _tellerV2, address _lenderCommitmentForwarder) {
+        TELLER_V2 = ITellerV2(_tellerV2);
+        LENDER_COMMITMENT_FORWARDER = ILenderCommitmentForwarder(
+            _lenderCommitmentForwarder
+        );
+    }
+
+
+    function rolloverLoan(
+        uint256 _loanId,
+        AcceptCommitmentArgs calldata _commitmentArgs
+    ) external returns (uint256 newLoanId_) {
+        address borrower = TELLER_V2.getLoanBorrower(_loanId);
+        require(borrower == msg.sender, "CommitmentRolloverLoan: not borrower");
+
+        // Get lending token and balance before
+        IERC20Upgradeable lendingToken = IERC20Upgradeable(
+            TELLER_V2.getLoanLendingToken(_loanId)
+        );
+        uint256 balanceBefore = lendingToken.balanceOf(address(this));
+
+        // Accept commitment and receive funds to this contract
+        newLoanId_ = _acceptCommitment(_commitmentArgs);
+
+        // Calculate funds received
+        uint256 fundsReceived = lendingToken.balanceOf(address(this)) -
+            balanceBefore;
+
+        // Approve TellerV2 to spend funds and repay loan
+        lendingToken.approve(address(TELLER_V2), fundsReceived);
+        TELLER_V2.repayLoanFull(_loanId);
```

Comment:
we will also need to account for any leftover tokens after the loan is repaid and send them to the borrower